From a9571102708bd7ebcd80a7ea271cca775013d97f Mon Sep 17 00:00:00 2001
From: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>
Date: Mon, 9 Sep 2024 14:09:13 +0200
Subject: [PATCH 10/10] Enable JITLink by default for AArch64 on Linux

There are still failures with RuntimeDyld on AArch64 which seem to
be solved with JITLink, so make the switch now.
---
 lib/Interpreter/IncrementalJIT.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Interpreter/IncrementalJIT.cpp b/lib/Interpreter/IncrementalJIT.cpp
index 98110b36..abd7b4ff 100644
--- a/lib/Interpreter/IncrementalJIT.cpp
+++ b/lib/Interpreter/IncrementalJIT.cpp
@@ -443,7 +443,8 @@ static bool UseJITLink(const Triple& TT) {
   if (TT.getArch() == Triple::riscv64 ||
       (TT.isOSBinFormatMachO() &&
        (TT.getArch() == Triple::aarch64 || TT.getArch() == Triple::x86_64)) ||
-      (TT.isOSBinFormatELF() && TT.getArch() == Triple::ppc64le)) {
+      (TT.isOSBinFormatELF() &&
+       (TT.getArch() == Triple::aarch64 || TT.getArch() == Triple::ppc64le))) {
     jitLink = true;
   }
   // Finally, honor the user's choice by setting an environment variable.
